changelog.d/15464.bugfix

```diff
@@ -0,0 +1 @@
+Don't fail moving read marker if current set to a missing event. Contributed by Nick @ Beeper (@fizzadar).
```

Comment:
I think this is description might help server admins understand better what this is fixing? What do you think?

```suggestion
Fix a long-standing bug where setting the read marker could fail when using message retention. Contributed by Nick @ Beeper (@fizzadar).
```